### PR TITLE
Switch to non-deprecated plugin options

### LINKED ISSUE #134: Deprecation messages in Vite 5
After upgrading to Vite 5, I see the following warning messages when starting the Vite server or building with Vite:

```
 WARN  plugin 'vite:html' uses deprecated 'enforce' option. Use 'order' option instead.
 WARN  plugin 'vite:html' uses deprecated 'transform' option. Use 'handler' option instead.
```

### PR DESCRIPTION
**Pull request by marvinruder**
* Update vite to a version supporting current plugin options

Fixes #134: Deprecation messages in Vite 5

### CODE REVIEW

**Comment by marvinruder** [comment]
Not sure if this change retains compatibility with all versions of Vite matching the `peerDependencies` range, so that range might need updating as well.

**Review by atom-git — APPROVED** [review]

**Comment by haiweilian** [comment]
我建议与所有版本兼容。这只是内部 API 更改，安装时我不需要注意版本。

I recommend compatibility with all versions. This is an internal API change only，I don't need to pay attention to the version when installing.

File: packages/core/src/htmlPlugin.ts
Comment (oinochoe) [inline packages/core/src/htmlPlugin.ts:1]: Thank you.
I think it will be okay with that option modified.

**Review by oinochoe — APPROVED** [review]

**Comment by marvinruder** [comment]
Closing in favor of #137.

**Pull request closed without merging**